pkg/cmd/server/start/start_master.go

```diff
@@ -475,16 +475,23 @@ func (m *Master) Start() error {
 			// continuously run the scheduler while we have the primary lease
 			go runEmbeddedScheduler(m.config.MasterClients.OpenShiftLoopbackKubeConfig, m.config.KubernetesMasterConfig.SchedulerConfigFile, m.config.KubernetesMasterConfig.SchedulerArguments)
 
+			controllerArgumentsInitialized := make(chan struct{})
+
 			go runEmbeddedKubeControllerManager(
 				m.config.MasterClients.OpenShiftLoopbackKubeConfig,
 				m.config.ServiceAccountConfig.PrivateKeyFile,
 				m.config.ServiceAccountConfig.MasterCA,
 				m.config.KubernetesMasterConfig.PodEvictionTimeout,
 				m.config.VolumeConfig.DynamicProvisioningEnabled,
 				m.config.KubernetesMasterConfig.ControllerArguments,
+				controllerArgumentsInitialized,
 				recyclerImage,
 				informers)
 
+			// Wait for the runEmbeddedKubeControllerManager to mutate the
+			// ControllerArguments to prevent getOpenshiftControllerOptions from
+			// panicking because of concurrent read/write access.
+			<-controllerArgumentsInitialized
```

Comment:
this smells. Do the openshift controller really have to see the mutated flags? Or was this only an accident?